### DISCUSSION TITLE
Adding "MobileAds" to Singleton

### DISCUSSION THREAD

**Discussion by tatarinovvladimir** [Poing-Studios/Godot-AdMob-Android-iOS#69] [Q&A]
Greetings.
Firstly, I would like to thank you for the admob module for iOS. The installation manual indicates that you need to add the MobileAds scene to startup. Since I use two different modules for advertising Shin-nel for android (since there you can put personalized / non-personalized ads directly from the application, depending on the user's choice) and yours for aios, I have created a self-written API to manage these modules and it is it that is added to startup. The problem is that both modules work on Android (not at the same time, of course, depending on which one you export with), and on iOS, the application does not see the API in startup and all advertising calls go to the Null object.

Question: Can your module work on iOS if MobileAds.tscn is not added to autoload, but is called through 
another script?

![image](https://user-images.githubusercontent.com/25502499/117512586-93883880-af98-11eb-900b-957e46ecb9ea.png)

The photo shows the structure. Both modules work through the Ads node, which is a common API for both modules.
On android it all works fine, but not on iOS.

**Comment by gumaciel** [comment]
Hi, i recommend you to use only one plugin/module (for AdMob), if you use more than one can cause a conflict.

This Plugin already support personalized ads, please read more here to how use the personalized ads on Android and iOS with UMP: https://support.google.com/admob/answer/7676680?hl=en

- Question: Can your module work on iOS if MobileAds.tscn is not added to autoload, but is called through
another script?

- Answer: Yes, but it's not recommended, read more here: https://github.com/Poing-Studios/Godot-AdMob-Android-iOS/wiki/Question-and-Answers#3-why-we-need-to-put-the-admob-api-on-autoload-and-singleton-and-not-add-on-every-scene

**Reply by tatarinovvladimir** [reply]
> Hi, i recommend you to use only one plugin/module (for AdMob), if you use more than one can cause a conflict.
> 
> This Plugin already support personalized ads, please read more here to how use the personalized ads on Android and iOS with UMP: https://support.google.com/admob/answer/7676680?hl=en
> 
> * Question: Can your module work on iOS if MobileAds.tscn is not added to autoload, but is called through
>   another script?
> * Answer: Yes, but it's not recommended, read more here: https://github.com/Poing-Studios/Godot-AdMob-Android-iOS/wiki/Question-and-Answers#3-why-we-need-to-put-the-admob-api-on-autoload-and-singleton-and-not-add-on-every-scene

I probably do not quite understand, your plugin can change personalized and non-personalized ads at runtime? It's just that your instructions indicate that personalized / non-personalized ads are installed on the admob website, but not directly from the application. If I can put personalized / non-personalized ads directly from the application with your plugin, then of course I would choose your plugin only.

**Reply by gumaciel** [reply]
As i know, the Shin-Nil Plugin does this: https://developers.google.com/admob/android/eu-consent#forward-consent, this is for EU consent and it's deprecated/legacy, we use this: https://developers.google.com/admob/ump/android/quick-start

And for this new EU consent we can't do that, only on [AdMob website](https://admob.google.com/):

![image](https://user-images.githubusercontent.com/20030153/117520938-772cd180-af81-11eb-8fe7-37d27062820f.png)

**Reply by tatarinovvladimir** [reply]
Well, at the moment, in that plugin, I can install a personalized or non-personalized ad just in Godot as set_personalized (true) right in the application. And all questions about the age of the user, etc., are done manually in the application. There is no way to manually set the parameter for personalized ads in your plugin.

**Reply by gumaciel** [reply]
Sorry about that, but we can't implement this via code because it will not work in our Plugin.

Understand that the way to tell if an ad is personalized or not directly by the code is deprecated/legacy, it may be that in the next versions this will stop working completely.

Also: This way of personalized ads by the AdMob `code or website` is only for users who are in the EEA, so if you are trying to use personalized ads without being in the EEA, it won't make a difference as far as I know.

**Reply by tatarinovvladimir** [reply]
I understood you. It turns out that then I can fully use your plugin. Thanks for the explanation of the situation.
